website/docs/expansion.md

```diff
@@ -0,0 +1,223 @@
+---
+id: expansion
+title: Early Rejection of Generator Resources
+---
+
+> 🚧 This feature is in _alpha_ stage, and is not enabled by default. To
+> enable, set the "enable-generator-resource-expansion" flag.
+
+Gatekeeper can be configured to reject generator resources that might create a
+resource that violates a policy. For example, one could configure Gatekeeper to
+immediately reject deployments that would create a `Pod` that violates a
+constraint instead of merely rejecting the Pods. To achieve this, Gatekeeper
+creates a "mock resource" for the `Pod`, runs validation on it, and aggregates
+the mock resource's violations onto the parent resource (the `Deployment` in
+this example).
+
+To use this functionality, first specify which resources should be "expanded" by
+creating 1 or more `TemplateExpansion` custom resources. Then, in order for
+Gatekeeper to accurately create these "mock resources" in a way that mirrors the
+real Kubernetes controllers, users can define any number of
+[Mutations](https://open-policy-agent.github.io/gatekeeper/website/docs/mutation)
```

Comment:
maybe mention something about for this expansion generator feature, we are leveraging Gatekeeper's mutation feature to create the mock resources.